aci/resource_aci_cloudtemplateextnetwork.go

```diff
@@ -0,0 +1,447 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/aci-go-client/v2/client"
+	"github.com/ciscoecosystem/aci-go-client/v2/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAciCloudTemplateforExternalNetwork() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAciCloudTemplateforExternalNetworkCreate,
+		UpdateContext: resourceAciCloudTemplateforExternalNetworkUpdate,
+		ReadContext:   resourceAciCloudTemplateforExternalNetworkRead,
+		DeleteContext: resourceAciCloudTemplateforExternalNetworkDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAciCloudTemplateforExternalNetworkImport,
+		},
+
+		SchemaVersion: 1,
+
+		CustomizeDiff: func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+			if diff.Get("cloud_vendor") != "gcp" {
```

Comment:
found this if statement hard to read because of the nested negative if statements

would something like this also work:

```
if diff.Get("cloud_vendor") != "gcp" && diff.Get("all_regions") == "no"{
	return fmt.Errorf("all_regions should always be set to yes when using %v cloud APIC", diff.Get("cloud_vendor"))
} else if diff.Get("cloud_vendor") == "gcp" && diff.Get("all_regions") == "yes" {
	return fmt.Errorf("all_regions should always be set to no when using GCP cloud APIC")
} else {
	return nil
}
```